frontend/public/components/environment.jsx

```diff
@@ -359,6 +371,17 @@ export const EnvironmentPage = connect(stateToProps)(
       }
     }
 
+    /**
+     * If we are using the editor in a create page, this function will export all the data
+     * needed to add the environment variables to the yaml when sending down the create
+     * action.
+     *
+     */
+    exportEnvironment() {
```

Comment:
Is it possible instead to have an optional `onChange` callback prop that returns the new values when they change? That seems a lot cleaner to me.